app/main.py

```diff
@@ -1,8 +1,38 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+        self.name = name
+        self.age = age
+
+        Person.people[self.name] = self
+
+
+def find_partner(person_obj: Person,
+                 person_to_find: str | None
+                 ) -> Person | str | None:
+    if person_to_find in Person.people:
+        partner = person_obj.people[person_to_find]
+        if hasattr(partner, "wife"):
+            partner.wife = person_obj
+        else:  # hasattr(partner, "husband"):
```

Comment:
There is no need for this comment, you could make the following condition
```suggestion
        elif hasattr(partner, "husband"):
```